pwiz_tools/Skyline/SkylineNightly/Xml.cs

```diff
@@ -38,19 +38,44 @@ private XmlNode GetRoot()
             return _rootNode;
         }
 
+        public static void ShowXmlExceptionDetails(Exception e, string hint)
+        {
+            var message = $@"XML exception ""{e.Message}"" \nat\n{e.StackTrace}\n while handling this text:\n{hint}";

```

Comment:
Usually you would throw an exception with just a little more information in its message and preserve "e" as the inner exception:
throw new Exception("XML exception while handling this text:\n"" + hint, e);

Is there a reason that you are trying to throw an exception with a big message and no nested exceptions?
Is the problem that somewhere else, all that is getting written to the log file is the message of the exception and no information about the inner exceptions?
Can we fix the other place so it outputs all of Exception.ToString()?

Also, if you do need to convert the inner exception to text it's best to use it's ToString() method because that walks down all of the nested exceptions and makes sure that the whole message contains ever exception's Message and StackTrace, and it even does the correct thing for "AggregateException".

Reply:
Excellent points, thanks. I'll tidy this up.

Reply:
Didn't properly answer your question, sorry - this is in response to SkylineNightly behavior like this:
![image](https://github.com/ProteoWizard/pwiz/assets/1200761/ad3d2bd1-5fe3-4986-9873-21e468449277)
just throwing that exception wasn't very helpful in tracking down the problem. The point here is definitely to get the trouble text onto the screen.

Reply:
Keeping in mind that it's going to appear in a system message box at program early exit. There's nothing to handle the rethrown exception.

Reply:
I think normally you would not put the stack trace in the Exception message. So:

var message = $@"XML exception while handling this text:\n{hint}";

Then the caller that catches this would be responsible for appropriately presenting both the stack trace of this throw and the stack trace of the embedded exception.

Reply:
There isn't anything catching this, though. This is SkylineNightly's dying gasp.  We could do something fancier but I don't think it's worth the effort. Though perhaps it should be called ShowXmlExceptionDetailsAndDie()

Reply:
So:

        // Caller isn't expected to handle this - these exceptions are rare, and this isn't
        // user facing so writing a bunch of error recovery code isn't a good use of time.
        public static void ShowXmlExceptionDetailsAndDie(Exception e, string hint)
        {
            var message = $@"XML exception while handling this text:\n{hint}";
            throw new Exception(message, e); 
        }